### ISSUE TITLE
PrometheusRule issues with Helm Chart 1.9.0

### ISSUE THREAD

**Issue by onedr0p** [rook/rook#10070]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

Getting an issue with `PrometheusJobMissing` and `PrometheusRuleFailures`:

<img width="1323" alt="image" src="https://user-images.githubusercontent.com/213795/163301555-eb7bd089-b529-48bb-b05b-d1e05869754c.png">

**Expected behavior:**

The rules to be correct and not be alerted

**How to reproduce it (minimal and precise):**

Upgraded to 1.9.0 and enabled prometheus rules in the rook-ceph-cluster chart

**Environment**:
* OS (e.g. from /etc/os-release): N/A
* Kernel (e.g. `uname -a`): N/A
* Cloud provider or hardware configuration: N/A
* Rook version (use `rook version` inside of a Rook Pod): 1.9.0
* Storage backend version (e.g. for ceph do `ceph -v`): ceph version 16.2.7 (dd0603118f56ab514f133c8d2e3adfc983942503) pacific (stable)
* Kubernetes version (use `kubectl version`): v1.23.5+k3s1
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): k3s
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest/ceph-toolbox.html)): HEALTH_OK

**Comment by bjw-s** [comment]
Just chiming in to let you know that I am seeing the exact same behavior

**Comment by budimanjojo** [comment]
I'm using the manifest in `deploy/example` directory and I have the same 'PrometheusJobMissing` alert but not `PrometheusRuleFailures`

**Comment by onedr0p** [comment]
The `PrometheusRuleFailures` seemed to have gone away after I cleaned up the old Prometheus rule.

`k delete prometheusrule prometheus-ceph-v16-rules -n rook-ceph`

**Comment by travisn** [comment]
For the PrometheusRuleFailures, I see now that it is coming from the `pools` group in the new rules here: https://github.com/rook/rook/blob/master/deploy/examples/monitoring/localrules.yaml#L727-L791

Is there a description that shows any more details? If one of these rules fails, there should be some other details such as the pools being nearly full.

**Comment by onedr0p** [comment]
In my prometheus logs this is the error I see that might be causing the `PrometheusRuleFailures`

```
prometheus-kube-prometheus-stack-prometheus-0 prometheus ts=2022-04-14T17:32:49.834Z caller=manager.go:610 level=warn component="rule manager" group=pools msg="Evaluating rule failed" rule="alert: CephPoolGrowthWarning\nexpr: (predict_linear(ceph_pool_percent_used[2d], 3600 * 24 * 5) * on(pool_id) group_right()\n  ceph_pool_metadata) >= 95\nlabels:\n  oid: 1.3.6.1.4.1.50495.1.2.1.9.2\n  severity: warning\n  type: ceph_default\nannotations:\n  description: |\n    Pool '{{ $labels.name }}' will be full in less than 5 days assuming the average fill-up rate of the past 48 hours.\n  summary: Pool growth rate may soon exceed it's capacity\n" err="found duplicate series for the match group {pool_id=\"1\"} on the left hand-side of the operation: [{container=\"mgr\", endpoint=\"http-metrics\", instance=\"10.69.38.88:9283\", job=\"rook-ceph-mgr\", namespace=\"rook-ceph\", pod=\"rook-ceph-mgr-a-6f58d6f75c-8lj72\", pool_id=\"1\", service=\"rook-ceph-mgr\"}, {container=\"mgr\", endpoint=\"http-metrics\", instance=\"10.69.38.78:9283\", job=\"rook-ceph-mgr\", namespace=\"rook-ceph\", pod=\"rook-ceph-mgr-a-69fb86584f-qqtgv\", pool_id=\"1\", service=\"rook-ceph-mgr\"}];many-to-many matching not allowed: matching labels must be unique on one side"
```

**Comment by bjw-s** [comment]
Agreed, that is the same error that I am seeing in Prometheus for the `CephPoolGrowthWarning` rule:

```console
found duplicate series for the match group {pool_id="1"} on the left hand-side of the operation: [{container="mgr", endpoint="http-metrics", instance="10.244.6.213:9283", job="rook-ceph-mgr", namespace="rook-ceph", pod="rook-ceph-mgr-b-84ff6f7f8-zzxps", pool_id="1", service="rook-ceph-mgr"}, {container="mgr", endpoint="http-metrics", instance="10.244.4.228:9283", job="rook-ceph-mgr", namespace="rook-ceph", pod="rook-ceph-mgr-b-84ff6f7f8-2fswt", pool_id="1", service="rook-ceph-mgr"}];many-to-many matching not allowed: matching labels must be unique on one side
```

**Comment by travisn** [comment]
@pcuzner Could you take a look at these issues?
1. In `PrometheusJobMissing`, where does the `ceph` job come from in this condition? It must not be created in rook deployments: `absent(up{job="ceph"})`
2. In `CephPoolGrowthWarning`, this condition seems to indicate the pool is going to fill, but it's unexpected. Any recommendations on troubleshooting?

**Comment by pcuzner** [comment]
@travisn presumably the alerts were taken verbatim, so the job name is the job name used in the cephadm environment that defines the scrape config. So this will be different in the context of rook
